### PR TITLE
Start Admin Page

### PR DESCRIPTION
**Pull request by jettdc**
Changes:
- Create Admin Page
- Add tabs on admin page
- Add list of account requests
- Add models for account requests
- Add ability to approve or deny account requests (both mock and real).
- Add database service functions for approvals and denials
- Add Post an announcement page with rich text editor
- Add database routes to post an announcement.
- Add blacklist tab
- Add models for blacklisted users
- Add ability to unblacklist users

### CODE REVIEW

**Comment by jettdc** [comment]
`/db/` is used for differentiating parts of the api, so all database endpoints are in that subcategory. My thought behind this was that we would also have things like `/email/` endpoints (or whatever other backend services we have eventually). Let me know if you think this is unnecessary and I will drop it in all cases. @nh602

**Comment by nh602** [comment]
> `/db/` is used for differentiating parts of the api, so all database endpoints are in that subcategory. My thought behind this  was that we would also have things like `/email/` endpoints (or whatever other backend services we have eventually). 

That makes sense to me, we should leave `/db/` then.

File: src/app/services/database-service/database.impl.service.ts [hunk 1]
```diff
@@ -38,4 +56,113 @@ export class DatabaseImplService implements DatabaseService {
         })
       );
   }
+
+  public getApplicants(): Observable<Applicant[]> {
+    return this.http
+      .get<Applicant[]>(`${environment.backendHost}/api/db/getApplicants?`)
```
[the remark this answers is not in the document]
Comment (jettdc) [reply src/app/services/database-service/database.impl.service.ts:62]: Fixed

File: src/app/services/database-service/database.impl.service.ts [hunk 2]
```diff
@@ -38,4 +56,113 @@ export class DatabaseImplService implements DatabaseService {
...
+      .pipe(
+        catchError((error) => {
+          console.log('Error getting blacklisted users.');
+          console.log(error);
+          return throwError(error);
+        })
+      );
+  }
+
+  public denyApplicant(
```
[the remark this answers is not in the document]
Comment (jettdc) [reply src/app/services/database-service/database.impl.service.ts:84]: So the only thing that might be kinda weird here is that the information that we are sending over is different.

**Review by nh602 — APPROVED** [review]

**Pull request merged**